src/Material/Menu.elm

```diff
@@ -0,0 +1,523 @@
+    Toggle oracle -> effect
+
+        (Effects.tick (\_ -> Open))
+
+        { model | open =
+                    case model.open of
+                      Nothing   -> Just False
+                      Just True -> Nothing
+                      _         -> model.open
+
+                , oracle =
+                    case model.open of
+                      Nothing   -> Just oracle
+                      Just True -> Just oracle
+                      _         -> model.oracle
+
+
+                , items =
+                    [1..List.length oracle.offsetTops]
+                    |> List.map (\i -> (i, Ripple.model))
+                    |> Dict.fromList
+        }
+
+    -- Transition from opening state to open state.
+    Open -> pure
+      { model | open = if model.open == Just False then
+                           Just True
+                         else
+                           model.open
+      }
+
+    -- Immediately close the menu.
+    Hide idx oracle -> effect
+
+      ( Effects.tick (\_ -> Select idx) )
+
+      { model | open = Nothing, closing = False, oracle = Just oracle }
+
+    -- Close the menu after some delay.
+    Close' idx oracle -> effect
+
+      ( Effects.task
+        <| Task.andThen (Task.sleep constant.closeTimeout) << always
+        <| Task.succeed (Hide idx oracle))
+
+      { model | closing = True }
+
+
+    Ripple idx action ->
+
+      let
+
+        (model', effects) = Dict.get idx model.items
+                         |> Maybe.withDefault Ripple.model
+                         |> Ripple.update action
+      in
+
+        effect
+
+          ( Effects.map (Ripple idx) effects )
+
+          { model | items = Dict.insert idx model' model.items }
+
+    Select _ -> pure model
+    Close -> pure model
+
+
+-- VIEW
+
+
+{-| Component view.
+-}
+view : Signal.Address Action -> Model -> List Item -> List Html
+view addr model items =
+  [ styled button
+      [ cs "mdl-button"
+      , cs "mdl-js-button"
+      , cs "mdl-button--icon"
+      , Style.attribute (onClick addr Oracle.decode Toggle)
+      ]
+      [ Html.i [ Html.class "material-icons" ] [ Html.text "more_vert" ]
+      ]
+  , styled div
+      [ cs "mdl-menu__container"
+      , cs "is-upgraded"
+      , cs' "is-visible" (model.open == Just True)
+
+      , css' "width"
+          ( case model.oracle of
+              Just oracle ->
+                oracle.menu.bounds.width
+                |> toString
+                |> flip (++) "px"
+              Nothing     -> "auto"
+          )
+          ( isJust model.oracle )
+
+      , css' "height"
+          ( case model.oracle of
+              Just oracle ->
+                oracle.menu.bounds.height
+                |> toString
+                |> flip (++) "px"
+              Nothing     -> "auto"
+          )
+          ( isJust model.oracle )
+
+      , flip (css' "top") (((model.alignment == bottomRight) || (model.alignment == bottomLeft)) && (isJust model.oracle)) <|
```

Comment:
This and the following looks to me like alignment ought to be a datatype, i.e., 
```
type Alignement 
  = BottomRight
  | BottomLeft 
  | TopRight
  | TopLeft
  | Unaligned
```
You can then use `case` to produce the `cs` you need.